docs/check-examples

```diff
@@ -126,7 +126,7 @@ do
   ## May not be robust against parentheses, brackets or other regex-special symbols in the error message -- none of these encountered so far.
 
   ( echo "$line" | grep -E -e '[#][#] ' > /dev/null ) \
-   && msg=`echo "$line" | sed -E -e 's/ \{#.*\}[ ]*$//' -e 's/^##[ ]*[*]*//' -e 's/[*]*$//' -e 's/\\*/\\\\*/' -e 's/_[a-z]+_/.*/g'`
+   && msg=`echo "$line" | sed -E -e 's/ \{#.*\}[ ]*$//' -e 's/^##[ ]*[*]*//' -e 's/[*]*$//' -e 's/\\*/\\\\*/' -e 's/\[/\\\[/g' -e 's/\]/\\\]/g' -e 's/_[a-z]+_/.*/g'`
```

Comment:
What is this change doing?

Reply:
The $line is an error message template as written in the documentation file.
The $msg is a regex that matches that template against actual error messages from examples in the text.
This change converts [ to \[ in the text so that the bracket is not seen as a regex-relevant symbol.